src/metrics/metrics_test.go

```diff
@@ -594,8 +675,14 @@ func TestPopulatePgBouncerMetrics(t *testing.T) {
 				"host":        "testhost",
 			}
 
-			mock.ExpectQuery("SHOW STATS;").
-				WillReturnRows(pgbouncerStatsRows)
+			if testCase.pgbouncerStatsRows == nil {
+				mock.ExpectQuery("SHOW STATS;").
+					WillReturnRows(pgbouncerStatsRows)
+			} else {
+				mock.ExpectQuery("SHOW STATS;").
+					WillReturnRows(testCase.pgbouncerStatsRows)
+			}
```

Comment:
Nit: we could update the previous test cases with:

```go
// ...
pgbouncerStatsRows: pgbouncerStatsRows,
// ...
```

That way we wouldn't need this if/else clasue, and the results would be explicit for each test case. We could even improve the `pgbouncerStatsRows` global variable name to say that these are the expected rows for versions prior to PgBouncer 23. Example: `pgbouncerPriorTo23StatsRows`.

Reply:
Makes sense. Will fix this.